### PR TITLE
fix premature exit when RG doesn't exist (#46013)

### LINKED ISSUE #45941: azure_rm_deployment failing to create resource group that doesn't pre-exist
##### SUMMARY

Creating an Azure Resource Group via ARM template using `azure_rm_deployment` fails in Ansible 2.6 if the resource group does not already exist. The reason this is is because `self.get_resource_group` does not exist https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/azure/azure_rm_deployment.py#L494 pre-Ansible 2.6 when the resource group hasn't been created yet.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

`azure_rm_deployment`
`azure_rm_common`

##### ANSIBLE VERSION

```paste below
ansible 2.6.4
  config file = /home/admin_user/.ansible.cfg
  configured module search path = [u'/home/admin_user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/admin_user/.ansible-virtualenv/lib/python2.7/site-packages/ansible
  executable location = /home/admin_user/.ansible-virtualenv/bin/ansible
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

`CentOS Linux release 7.5.1804 (Core)`

##### STEPS TO REPRODUCE

```yaml
- name: Load the ARM Template
  include_vars:
    file: "/tmp/{{ azure_iaas_deployment_name }}.yml.j2"
    name: az_template

- name: Create Azure Deployment
  azure_rm_deployment:
    state: "{{ azure_iaas_state }}"
    deployment_name: "{{ azure_iaas_deployment_name }}"
    resource_group_name: "{{ azure_iaas_resourcegroup }}"
    location: "{{ az.defaultRegion }}"
    template: "{{ az_template }}"
    client_id: "{{ az_client_id }}"
    subscription_id: "{{ az.subscription }}"
    tenant: "{{ az_tenant }}"
    secret: "{{ az_client_secret }}"
    tags: "{{ az_default_tags }}"
```

##### EXPECTED RESULTS

Azure resource group created.

##### ACTUAL RESULTS

```paste below
TASK [ansible-role-azure-iaas : Create Azure Deployment] **********************************************************************************************************************************************************************************************************************
task path: /home/smgv/Ansible-IAAS-VM/roles/ansible-role-azure-iaas/tasks/main.yml:111
Thursday 20 September 2018  19:54:46 +0000 (0:00:00.048)       0:00:29.565 ****
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: smgv
<127.0.0.1> EXEC /bin/sh -c 'echo ~smgv && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/smgv/.ansible/tmp/ansible-tmp-1537473286.99-252170810387720 `" && echo ansible-tmp-1537473286.99-252170810387720="` echo /home/smgv/.ansible/tmp/ansible-tmp-1537473286.99-252170810387720 `" ) && sleep 0'
Using module file /home/smgv/.ansib2.6/lib/python2.7/site-packages/ansible/modules/cloud/azure/azure_rm_deployment.py
<127.0.0.1> PUT /home/smgv/.ansible/tmp/ansible-local-9733759Ve9n/tmpiqgIpi TO /home/smgv/.ansible/tmp/ansible-tmp-1537473286.99-252170810387720/azure_rm_deployment.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/smgv/.ansible/tmp/ansible-tmp-1537473286.99-252170810387720/ /home/smgv/.ansible/tmp/ansible-tmp-1537473286.99-252170810387720/azure_rm_deployment.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/smgv/.ansib2.6/bin/python2 /home/smgv/.ansible/tmp/ansible-tmp-1537473286.99-252170810387720/azure_rm_deployment.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/smgv/.ansible/tmp/ansible-tmp-1537473286.99-252170810387720/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
  File "/tmp/ansible_ikYD7J/ansible_modlib.zip/ansible/module_utils/azure_rm_common.py", line 518, in get_resource_group
    return self.rm_client.resource_groups.get(resource_group)
  File "/home/smgv/.ansib2.6/lib/python2.7/site-packages/azure/mgmt/resource/resources/v2017_05_10/operations/resource_groups_operations.py", line 287, in get
    raise exp

fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "deployment_mode": "incremental",
            "deployment_name": "azure-iaas-2018-09-20T19-54-42.748577Z",
            "location": "southcentralus",
            "parameters": null,
            "parameters_link": null,
            "password": null,
            "profile": null,
            "resource_group_name": "smgv-s19-test26",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "present",
            "subscription_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "tags": {
                "CompanyCode": "12345",
                "CostElement": "12345",
                "Environment": "dev",
                "OwnerEmail": " team@company.com",
                "ServiceID": "12345"
            },
            "template": {
                "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
                "contentVersion": "1.0.0.0",
                "resources": [
                    {
                        "apiVersion": "2017-03-30",
                        "copy": {
                            "count": "[variables('vmCount')]",
                            "name": "vmLoop"
                        },
                        "dependsOn": [
                            "[resourceId('Microsoft.Network/networkInterfaces', concat(variables('vm_name'), padLeft(copyindex(),2,'0'), '-nic'))]"
                        ],
                        "location": "[variables('location')]",
                        "name": "[concat(variables('vm_name'), padLeft(copyindex(),2,'0'))]",
                        "properties": {
                            "hardwareProfile": {
                                "vmSize": "Standard_DS3_v2"
                            },
                            "licenseType": "Windows_Server",
                            "networkProfile": {
                                "networkInterfaces": [
                                    {
                                        "id": "[resourceId('Microsoft.Network/networkInterfaces', concat(variables('vm_name'), padLeft(copyindex(),2,'0'), '-nic'))]"
                                    }
                                ]
                            },
                            "osProfile": {
                                "adminPassword": "PASSWORD",
                                "adminUsername": "scotttest",
                                "computerName": "[concat(variables('vm_name'), padLeft(copyindex(),2,'0'))]"
                            },
                            "storageProfile": {
                                "dataDisks": [
                                    {
                                        "caching": "ReadOnly",
                                        "createOption": "Empty",
                                        "diskSizeGB": 64,
                                        "lun": 0,
                                        "managedDisk": {
                                            "storageAccountType": "Premium_LRS"
                                        },
                                        "name": "[concat(variables('vm_name'), padLeft(copyindex(),2,'0'), 'lun', '0')]"
                                    },
                                    {
                                        "caching": "none",
                                        "createOption": "Empty",
                                        "diskSizeGB": 64,
                                        "lun": 1,
                                        "managedDisk": {
                                            "storageAccountType": "Premium_LRS"
                                        },
                                        "name": "[concat(variables('vm_name'), padLeft(copyindex(),2,'0'), 'lun', '1')]"
                                    }
                                ],
                                "imageReference": {
                                    "id": "[variables('image_id')]"
                                },
                                "osDisk": {
                                    "caching": "ReadWrite",
                                    "createOption": "FromImage",
                                    "managedDisk": {
                                        "storageAccountType": "Premium_LRS"
                                    },
                                    "name": "[concat(variables('vm_name'), padLeft(copyindex(),2,'0'),'-osdisk')]",
                                    "osType": "Windows"
                                }
                            }
                        },
                        "resources": [
                            {
                                "apiVersion": "2015-06-15",
                                "dependsOn": [
                                    "[concat('Microsoft.Compute/virtualMachines/', variables('vm_name'), padLeft(copyindex(),2,'0'))]"
                                ],
                                "location": "[resourceGroup().location]",
                                "name": "CustomScriptExtension",
                                "properties": {
                                    "autoUpgradeMinorVersion": true,
                                    "publisher": "Microsoft.Compute",
                                    "settings": {
                                        "commandToExecute": "powershell.exe -ExecutionPolicy Unrestricted -File ./ConfigureRemotingForAnsible.ps1 -Verbose",
                                        "fileUris": [
                                            "https://artifactory.azure.company.com:443//iso/ConfigureRemotingForAnsible.ps1"
                                        ]
                                    },
                                    "type": "CustomScriptExtension",
                                    "typeHandlerVersion": "1.8"
                                },
                                "type": "extensions"
                            },
                            {
                                "apiversion": "2015-06-15",
                                "dependsOn": [
                                    "[concat('Microsoft.Compute/virtualMachines/', variables('vm_name'), padLeft(copyindex(),2,'0'))]"
                                ],
                                "location": "[resourceGroup().location]",
                                "name": "[concat(variables('vm_name'), copyindex(), '-iaasantimalware')]",
                                "properties": {
                                    "autoupgrademinorversion": true,
                                    "publisher": "Microsoft.Azure.Security",
                                    "settings": {
                                        "AntimalwareEnabled": true,
                                        "Exclusions": {
                                            "extensions": null,
                                            "paths": null,
                                            "processes": null
                                        },
                                        "realtimeprotectionenabled": true,
                                        "scheduledscansettings": {
                                            "day": 7,
                                            "isenabled": true,
                                            "scantype": "quick",
                                            "time": 120
                                        }
                                    },
                                    "type": "iaasantimalware",
                                    "typeHandlerVersion": "1.3"
                                },
                                "type": "extensions"
                            }
                        ],
                        "scale": null,
                        "tags": {
                            "AppRole": "app",
                            "CompanyCode": "12345",
                            "CostElement": "12345",
                            "Environment": "dev",
                            "OwnerEmail": " team@company.com",
                            "ServiceID": "12345"
                        },
                        "type": "Microsoft.Compute/virtualMachines"
                    },
                    {
                        "apiVersion": "2017-06-01",
                        "copy": {
                            "count": "[variables('vmCount')]",
                            "name": "nicLoop"
                        },
                        "dependsOn": [],
                        "location": "[variables('location')]",
                        "name": "[concat(variables('vm_name'),padLeft(copyindex(),2,'0'), '-nic')]",
                        "properties": {
                            "enableAcceleratedNetworking": true,
                            "ipConfigurations": [
                                {
                                    "name": "ipconfig1",
                                    "properties": {
                                        "primary": true,
                                        "privateIPAddressVersion": "IPv4",
                                        "privateIPAllocationMethod": "Dynamic",
                                        "subnet": {
                                            "id": "[variables('subnet1Ref')]"
                                        }
                                    }
                                }
                            ],
                            "primary": true,
                            "virtualMachine": {
                                "id": "[resourceId('Microsoft.Compute/virtualMachines', concat(variables('vm_name'), padLeft(copyindex(),2,'0')))]"
                            }
                        },
                        "tags": {
                            "AppRole": "app",
                            "CompanyCode": "12345",
                            "CostElement": "12345",
                            "Environment": "dev",
                            "OwnerEmail": " team@company.com",
                            "ServiceID": "12345"
                        },
                        "type": "Microsoft.Network/networkInterfaces"
                    }
                ],
                "variables": {
                    "as_name": "[concat(variables('vm_name'), '-as')]",
                    "image_id": "/subscriptions/********/resourceGroups/rgImageBuild/providers/Microsoft.Compute/images/server-2016-base-ussc",
                    "location": "southcentralus",
                    "subnet1Ref": "[concat(variables('vnetID'),'/subnets/', variables('virtualSubnetName'))]",
                    "virtualNetworkName": "default-ussc-s19us",
                    "virtualNetworkResourceGroup": "network-ussc-s19us",
                    "virtualSubnetName": "default-10.71.238.0_24",
                    "vmCount": 1,
                    "vm_name": "lindrv-dapp",
                    "vnetID": "[resourceId(variables('virtualNetworkResourceGroup'), 'Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]"
                }
            },
            "template_link": null,
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "wait_for_deployment_completion": true,
            "wait_for_deployment_polling_period": 10
        }
    },
    "msg": "Error retrieving resource group smgv-s19-test26 - Resource group 'smgv-s19-test26' could not be found."
```

### PR DESCRIPTION
**Pull request by nitzmahone**
##### SUMMARY
(backport #46013 to stable-2.7)

* fixes #45941
* corrects regression introduced by #26104; when the resource group doesn't exist, the module exits prematurely with an error instead of creating it.

(cherry picked from commit 3b52d968e678a9055a5b3e38185c889e1ddc027f)

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
azure_rm_deployment

##### ANSIBLE VERSION
2.7.0rc3

### CODE REVIEW

**Review by yungezz — APPROVED** [review]

**Comment by Fred-sun** [comment]
@nitzmahone Could you help confirm this PR should be merged or need others to review? Thanks!

**Comment by abadger** [comment]
Merged for the 2.7.1 release.

@Fred-sun We were waiting on the release of 2.7.0 since we were at rc4 before merging more non-blocker changes into the stable-2.7 branch.  Apologies for not communicating that well before.

**Pull request merged**
